benchmarks/graphics_pipeline/main.cpp

```diff
@@ -189,6 +230,159 @@ void ProjApp::Setup()
+        }
+
+        // Sampler
+        grfx::SamplerCreateInfo samplerCreateInfo = {};
+        samplerCreateInfo.magFilter               = grfx::FILTER_LINEAR;
+        samplerCreateInfo.minFilter               = grfx::FILTER_LINEAR;
+        samplerCreateInfo.mipmapMode              = grfx::SAMPLER_MIPMAP_MODE_LINEAR;
+        samplerCreateInfo.minLod                  = 0;
+        samplerCreateInfo.maxLod                  = FLT_MAX;
+        PPX_CHECKED_CALL(GetDevice()->CreateSampler(&samplerCreateInfo, &mSampler));
+    }
+
+    // Descriptor
+    {
+        grfx::DescriptorPoolCreateInfo poolCreateInfo = {};
+        poolCreateInfo.uniformBuffer                  = 8;
+        poolCreateInfo.sampledImage                   = 8;
+        poolCreateInfo.sampler                        = 8;
+        PPX_CHECKED_CALL(GetDevice()->CreateDescriptorPool(&poolCreateInfo, &mDescriptorPool));
+
+        // Descriptor set layout for buffers
+        {
+            grfx::DescriptorSetLayoutCreateInfo layoutCreateInfo = {};
+            layoutCreateInfo.flags.bits.pushable                 = true;
+            layoutCreateInfo.bindings.push_back(grfx::DescriptorBinding(0, grfx::DESCRIPTOR_TYPE_UNIFORM_BUFFER));
+            PPX_CHECKED_CALL(GetDevice()->CreateDescriptorSetLayout(&layoutCreateInfo, &mDescriptorSetLayoutBuffers));
+        }
+
+        // Descriptor set layout for textures and samplers
+        {
+            grfx::DescriptorSetLayoutCreateInfo layoutCreateInfo = {};
+            layoutCreateInfo.bindings.push_back(grfx::DescriptorBinding(1, grfx::DESCRIPTOR_TYPE_SAMPLED_IMAGE));
+            layoutCreateInfo.bindings.push_back(grfx::DescriptorBinding(2, grfx::DESCRIPTOR_TYPE_SAMPLER));
+            PPX_CHECKED_CALL(GetDevice()->CreateDescriptorSetLayout(&layoutCreateInfo, &mDescriptorSetLayout));
+        }
+
+        // Allocate descriptor set and write descriptors
+        {
+            PPX_CHECKED_CALL(GetDevice()->AllocateDescriptorSet(mDescriptorPool, mDescriptorSetLayout, &mSkyBoxDescriptorSet));
+
+            grfx::WriteDescriptor write = {};
+            // SkyBox
+            write            = {};
+            write.binding    = 1;
+            write.type       = grfx::DESCRIPTOR_TYPE_SAMPLED_IMAGE;
+            write.pImageView = mSampledImageView;
+            PPX_CHECKED_CALL(mSkyBoxDescriptorSet->UpdateDescriptors(1, &write));
+
+            // Sampler
+            write          = {};
+            write.binding  = 2;
+            write.type     = grfx::DESCRIPTOR_TYPE_SAMPLER;
+            write.pSampler = mSampler;
+            PPX_CHECKED_CALL(mSkyBoxDescriptorSet->UpdateDescriptors(1, &write));
+        }
+    }
+
+    // Entities
+    {
+        // SkyBox
+        TriMesh mesh = TriMesh::CreateCube(float3(1, 1, 1), TriMeshOptions().TexCoords());
+        SetupEntity(mesh, GeometryOptions::InterleavedU16().AddTexCoord(), &mSkyBox);
+
+        // Sphere
```

Comment:
how do you plan to implement the LODs for the spheres? you could find more details in the Knob design doc about the LODs for the model

Reply:
I haven't thought about the LODs for the spheres yet, so I was planning to finish this first part and then work on LODs.

Reply:
sure, make sense